src/jaegertracing/CMakeLists.txt

```diff
@@ -0,0 +1,195 @@
+# Build the jaegertracing library its self
+cmake_minimum_required(VERSION 3.3)
+
+project(libjaegertracing VERSION ${jaegertracing_VERSION} LANGUAGES CXX)
```

Comment:
This seems like a bad idea. It should all be one project. Not sure why splitting at all TBH.

Reply:
Separate project: convenience and seems to be standard practice with cmake, but I don't feel strongly about it.

Splitting at all: readability, maintainability, separation of concerns. The tests don't intermix with the main library and trample on its settings, etc.

See also https://rix0r.nl/blog/2015/08/13/cmake-guide/ and many others.

If I land up adopting this seriously I'll probably send a followup pull to extract the test sources into a separate subtree and ensure there's zero inclusion of test sources by non-test code, really make the boundary clear. I'll also submit another to pull the thrift generated code out of the source tree and build it on demand, something that'll be much easier with the possible fix to #45 I've been looking at.

Reply:
I have seen splitting the `CMakeLists.txt` file into multiple subdirectories, but I guess not subprojects. I see it is allowed in the link you pointed out, just hope it doesn't mess with other dependencies, etc.

Reply:
OK, I'll remove the subproject

Reply:
Thanks